kvirt/k3s/__init__.py

```diff
@@ -72,11 +73,21 @@ def create(config, plandir, cluster, overrides):
             else:
                 error("You need to define api_ip in your parameters file")
                 os._exit(1)
-    install_k3s_channel = data.get('install_k3s_channel', 'stable')
-    if install_k3s_channel not in ['stable', 'latest', 'testing']:
-        error("Invalid K3S install channel %s" % install_k3s_channel)
-        os._exit(1)
     data['basedir'] = '/workdir' if os.path.exists("/i_am_a_container") else '.'
+    install_k3s_args = []
+    for arg in data:
+        if arg.startswith('install_k3s'):
+            install_k3s_args.append("%s=%s" % (arg.upper(), data[arg]))
+    if not install_k3s_args:
```

Comment:
Could this be simpler? As the nice implementation of support K3s install variables supports them all. We can avoid having the `channel` and `version` parameters. Especially now that they don't map 1<>1 to the K3s ones. As that also requires extra code to handle if `if not install_k3s_args:` is ==> `true`. You already have the freedom - this suggested implementation of all K3s install vars.

Reply:
what do you mean with simpler?

Reply:
I've edited my comment. I hope it's more clear now.

By simple I mean only having more ways to configure something if it makes clear-cut sense. In this case I think it risks muddying the picture of how to customize the install of K3s with KCLI.
One doesn't really need the `channel` and `version` *KCLI* parameters as they can be set via the *KCLI* support for the `arg.startswith('install_k3s')` approach.

Reply:
ok, i hear you. Removed version and channel and just made sure that if no install_k3s_* parameter is provided, we set the channel to stable